server/types/oid/iterate.go

```diff
@@ -152,9 +153,7 @@ func IterateDatabase(ctx *sql.Context, database string, callbacks Callbacks) err
 		}
 	}
 
-	doltSession := dsess.DSessFromSess(ctx.Session)
-
-	cat := sqle.NewDefault(doltSession.Provider()).Analyzer.Catalog
+	cat := sqlserver.GetRunningServer().Engine.Analyzer.Catalog
```

Comment:
Interesting, wasn't aware that this existed

Reply:
Yeah, kinda dirty with the global and all, but I noticed in the profiler that a lot of time was being spent creating a new engine, just to grab the catalog.